core/server/master/src/main/java/alluxio/master/metastore/caching/CachingInodeStore.java

```diff
@@ -659,10 +662,13 @@ public void removeEdge(Edge edge) {
      * Gets all children of an inode, falling back on the edge cache if the listing isn't cached.
      *
      * @param inodeId the inode directory id
+     * @param option the read options
      * @return the ids of all children of the directory
      */
-    public Collection<Long> getChildIds(Long inodeId) {
-      evictIfNecessary();
+    public Collection<Long> getChildIds(Long inodeId, ReadOption option) {
```

Comment:
They are most likely entries that are loaded with shouldSkipCache set to false. A read operation suppressing eviction for other entries is not safe as in, we could end up with full cache. I think we shouldn't interfere with the eviction of entries that are already in the cache, and just avoiding cache should be good enough. WDYT?

Reply:
The idea of `skipCache` is to not interfere with the caching and eviction behavior between cache and backing store. For operations with `skipCache=false`, they will evict the cache if it's full. Think of it this way:  the caching inode store works normally for operations other than scan, for scan, it will neither cache nor evict, so scan a caching inode store will have no effect on the caching status of the store.

Reply:
During scanning the nodes that are evicted are brought by it, hence the trashing. I'm in for keeping the logic simple and only skip caching. That will already prevent trashing since cache won't be filled by scanning and eviction won't be necessary due to scanning.

Reply:
SG, since the eviction thread keeps running in background, the "skip eviction" does not make much sense.

Reply:
Eviction here is in-memory. So no effect on trashing as well.